ksql-engine/src/main/java/io/confluent/ksql/function/UdfLoader.java

```diff
@@ -203,8 +198,7 @@ private ClassAnnotationMatchProcessor handleUdafAnnotation(final ClassLoader loa
                   udafAnnotation.name(),
                   path,
                   method.getDeclaringClass());
-              return Optional.of(compiler.compileAggregate(method,
-                  loader,
+              return Optional.of(createUdafFactoryInvoker(method,
```

Comment:
Not passing the class loader down seems wrong, though I'm no expert on this code.

If memory serves me right we're using custom class loaders when loading UDFs.  It might be that our tests pass with this change, but it might fail for someone implementing their own UDF jar.  Agreed that this would be a case of a missing important test, but I think we need to be careful we're not breaking something here.

cc @dguy who might be able to add more.

Reply:
Thanks @dguy - I'm wondering if we should have a test for this!

Reply:
`UDFLoaderTest.shouldPutJarUdfsInClassLoaderForJar` ?

Reply:
The classloaders _are_ still passed in and used for looking up udfs/udafs, it's just not passed to what was the compiler as we're not doing any compilation any more.

Reply:
The code that actually loads the udf/udaf classes from the provided classloader hasn't changed.

Reply:
Looks like that test passes even without passing around the class loader though.

Potentially, we need more of a functional test, i.e. a test around the use of an example UDF that won't work without the custom class loading...

Reply:
The way to get classloader isolation (e.g. so different UDFs could depend on different classes with the same names) is to use a custom classloader which use child first instead of parent first delegation. I don't know if that's the case in the code. But the code that loads the classes for the udf hasn't changed in this PR so whatever behaviour was there before should still be the same.
The actual loading seems to be done with a 3rd party class called `FastClasspathScanner` which I'm not familiar with, but it looks like we specify a classloader to use for loading classes with the `overrideClassLoaders` method. A new instance of this classloader is created for each file found in the plugins dir.

Reply:
> as I say, this code hasn't changed anyway.

Unresolved, as I'm confused.... I'm highlighting code that no longer passes around the class loader, so I'd say the code has changed!  

However, are you saying that because the `Method` instance coming from that other class loader that everything should be ok?

It still makes me nervous to merge this without the additional testing, given that if we have broken something it will be regression, and one not picked up until #3597 is done.

However, if @dguy is happy with the change, then I am.  Failing that, do you think we can add more testing around this, given the code _has_ changed?

Reply:
Unresolved, as I'm confused.... I'm highlighting code that no longer passes around the class loader, so I'd say the code has changed!

However, are you saying that because the Method instance coming from that other class loader that everything should be ok?

It still makes me nervous to merge this without the additional testing, given that if we have broken something it will be regression, and one not picked up until #3597 is done.

However, if @dguy is happy with the change, then I am. Failing that, do you think we can add more testing around this, given the code has changed?

Reply:
I will add a new test anyway. But I'm pretty sure the udf/udaf classloading code hasn't changed. The classloader that was being passed around after that was for the code generation (which no longer exists) not for the udf loading so shouldn't be relevant. But I will check.